src/posix/platform/netif.cpp

```diff
@@ -1598,41 +1604,54 @@ static void processMLDEvent(otInstance *aInstance)
 
 #if defined(__linux__)
 // set up the tun device
-static void platformConfigureTunDevice(const char *aInterfaceName, char *deviceName, size_t deviceNameLen)
+static void platformConfigureTunDevice(otPlatformConfig *aPlatformConfig)
 {
     struct ifreq ifr;
+    const char * interfaceName;
 
     sTunFd = open(OPENTHREAD_POSIX_TUN_DEVICE, O_RDWR | O_CLOEXEC | O_NONBLOCK);
     VerifyOrDie(sTunFd >= 0, OT_EXIT_ERROR_ERRNO);
 
     memset(&ifr, 0, sizeof(ifr));
-    ifr.ifr_flags = IFF_TUN | IFF_NO_PI | static_cast<short>(IFF_TUN_EXCL);
+    ifr.ifr_flags = IFF_TUN | IFF_NO_PI;
+    if (!aPlatformConfig->mPersistentInterface)
+    {
+        ifr.ifr_flags |= static_cast<short>(IFF_TUN_EXCL);
+    }
 
-    if (aInterfaceName)
+    interfaceName = aPlatformConfig->mInterfaceName;
+    if (interfaceName)
     {
-        VerifyOrDie(strlen(aInterfaceName) < IFNAMSIZ, OT_EXIT_INVALID_ARGUMENTS);
+        VerifyOrDie(strlen(interfaceName) < IFNAMSIZ, OT_EXIT_INVALID_ARGUMENTS);
 
-        strncpy(ifr.ifr_name, aInterfaceName, IFNAMSIZ);
+        strncpy(ifr.ifr_name, interfaceName, IFNAMSIZ);
     }
     else
     {
         strncpy(ifr.ifr_name, "wpan%d", IFNAMSIZ);
     }
 
     VerifyOrDie(ioctl(sTunFd, TUNSETIFF, static_cast<void *>(&ifr)) == 0, OT_EXIT_ERROR_ERRNO);
-    VerifyOrDie(ioctl(sTunFd, TUNSETLINK, ARPHRD_VOID) == 0, OT_EXIT_ERROR_ERRNO);
 
-    strncpy(deviceName, ifr.ifr_name, deviceNameLen);
+    strncpy(gNetifName, ifr.ifr_name, sizeof(gNetifName));
+
+    if (aPlatformConfig->mPersistentInterface)
+    {
+        VerifyOrDie(ioctl(sTunFd, TUNSETPERSIST, 1) == 0, OT_EXIT_ERROR_ERRNO);
+        SetLinkState(gInstance, false); // Set link down to reset the tun configuration.
```

Comment:
Could you add comments about what will be reset?

Reply:
Add more comments